### PR TITLE
Update snowplow

### PR DESCRIPTION
**Pull request by vauxhall**

### CODE REVIEW

File: Client/Ecosia/Analytics/Analytics.swift
Comment (ecotopian) [inline Client/Ecosia/Analytics/Analytics.swift:13]: Have you checked with BI whether we need these `platformContext ` and `geoLocationContext `? Have you also checked the other parameters like `exceptionAutotracking `, `installAutotracking` or `lifecycleAutotracking`? just to name a few. Seems there's lots of new stuff that's automatically tracked 🤔 

I think most are [true](https://snowplow.github.io/snowplow-objc-tracker/interface_s_p_tracker_configuration.html) by default, but actually we should set them to false as we don't need those events and can save the bandwidth, don't we?

File: Client/Ecosia/Analytics/Analytics.swift
Comment (ecotopian) [inline Client/Ecosia/Analytics/Analytics.swift:27]: maybe we can remove this if `installAutotracking ` is enabled? otherwise it could turn out we count installs twice if auto tracking uses the same schema (what might be)

File: Client/Ecosia/Analytics/Analytics.swift
Comment (vauxhall) [reply Client/Ecosia/Analytics/Analytics.swift:13]: @ecotopian No I didn't check with BI. My scope here is to update to the latest SDK and leave it in a state where it can work as before.

But the points you make are worth exploring and maybe we should set a meeting with BI to see if they have more insights on it, maybe is also worth it to create a separate ticket for that.

File: Client/Ecosia/Analytics/Analytics.swift
Comment (vauxhall) [reply Client/Ecosia/Analytics/Analytics.swift:27]: @ecotopian As far as I remember we needed to set our own install event as on the BI side they could not read it directly from the auto tracked one. I remember a conversation around it last year, but there is no documentation and better don't take my word for it. This should also be on a separated _optimisation_ ticket.

File: Client/Ecosia/Analytics/Analytics.swift
Comment (ecotopian) [reply Client/Ecosia/Analytics/Analytics.swift:13]: Yes, good idea! I'll approach them to find out more about the changes.

**Review by ecotopian — APPROVED** [review]
📈 looks great!

**Comment by ecotopian** [comment]
Let's not merge before we have green light from BI, but it's great that it's there 👍

**Comment by vauxhall** [comment]
@ecotopian I need to changes on this PR for the feature I'm working on. I don't want to merge this branch into main and later rebase because that means fixing merge conflicts again.

I propose to merge this and when we get feedback from BI we adjust if necessary.

**Comment by ecotopian** [comment]
@vauxhall  OK

**Pull request merged**
